plugins/outputs/riemann/riemann.go

```
@@ -68,8 +68,13 @@ func (r *Riemann) Write(metrics []telegraf.Metric) error {
 
 	var senderr = r.client.SendMulti(events)
 	if senderr != nil {
-		return errors.New(fmt.Sprintf("FAILED to send riemann message: %s\n",
-			senderr))
+		r.Close()
```

you shouldn't ignore errors from the Close and Connect functions